cmd/starbridge/integrations/ethereum.go

```diff
@@ -47,41 +48,60 @@ func Ethereum2Transaction(conn *ethclient.Client, txReceipt *types.Receipt, tx *
 	if !IsMyContractAddress(tx.To().Hex()) {
 		return nil, fmt.Errorf("unsupported receiver address '%s'", tx.To().Hex())
 	}
-
 	fromAddress, e := getFromAddress(tx)
 	if e != nil {
 		return nil, fmt.Errorf("unable to get From address: %s", e)
 	}
 
-	counter, e := Counter.NewCounter(common.HexToAddress(MY_ETHEREUM_CONTRACT_ADDRESS), conn)
-	if e != nil {
-		return nil, fmt.Errorf("unable to construct counter: %s", e)
+	// pulled logic to read events from here: https://goethereumbook.org/event-read/ with many modifications to suit our smart contract
```

Comment:
new way of querying smart contract because we use events. It is much easier now with explicit events